src/core/backend/pearl/group.rs

```diff
@@ -342,3 +358,40 @@ impl Display for Group {
             .finish()
     }
 }
+
+struct PartitionName {
+    timestamp: u64,
+    hash: String,
+}
+
+impl PartitionName {
+    fn new(timestamp: u64, hash: &str) -> Self {
+        Self {
+            timestamp,
+            hash: hash.to_string(),
+        }
+    }
+
+    fn try_from_string(s: &str) -> Option<Self> {
+        let (timestamp_string, hash_string) = if let Some(index) = s.find('_') {
```

Comment:
You can use [str.split](https://doc.rust-lang.org/std/primitive.str.html#method.split)